shared/src/api/extension/api/roots.ts

```diff
@@ -1,32 +1,44 @@
 import { ProxyMarked, proxyMarker } from 'comlink'
 import * as clientType from '@sourcegraph/extension-api-types'
-import { Subject } from 'rxjs'
+import { Subject, BehaviorSubject } from 'rxjs'
 import * as sourcegraph from 'sourcegraph'
 
 /** @internal */
-export interface ExtRootsAPI extends ProxyMarked {
+export interface ExtWorkspaceAPI extends ProxyMarked {
     $acceptRoots(roots: readonly clientType.WorkspaceRoot[]): void
+    $acceptVersionContext(versionContext: string | undefined): void
 }
 
 /** @internal */
-export class ExtRoots implements ExtRootsAPI, ProxyMarked {
+export class ExtWorkspace implements ExtWorkspaceAPI, ProxyMarked {
     public readonly [proxyMarker] = true
 
     private roots: readonly sourcegraph.WorkspaceRoot[] = []
+    private versionContext: string | undefined = undefined
 
     /**
      * Returns all workspace roots.
      *
      * @internal
      */
-    public getAll(): readonly sourcegraph.WorkspaceRoot[] {
+    public getAllRoots(): readonly sourcegraph.WorkspaceRoot[] {
         return this.roots
     }
 
-    public readonly changes = new Subject<void>()
+    public getVersionContext(): string | undefined {
+        return this.versionContext
+    }
+
+    public readonly rootsChanges = new Subject<void>()
```

Comment:
Double plural (this is also what it's called in the API)
```suggestion
    public readonly rootChanges = new Subject<void>()
```